suprb/optimizer/rule/mutation.py

```diff
@@ -69,47 +87,108 @@ class Normal(RuleMutation):
-
     def ordered_bound(self, rule: Rule, random_state: RandomState):
         bounds = rule.match.bounds
-        bounds[:, 0] -= self.mutation(dimensions=bounds.shape[0], random_state=random_state)
-        bounds[:, 1] += self.mutation(dimensions=bounds.shape[0], random_state=random_state)
+        bounds[:, 0] -= halfnorm.rvs(scale=self.sigma / 2, size=bounds.shape[0], random_state=random_state)
+        bounds[:, 1] += halfnorm.rvs(scale=self.sigma / 2, size=bounds.shape[0], random_state=random_state)
         rule.match.bounds = np.sort(rule.match.bounds, axis=1)
 
+    def unordered_bound(self, rule: Rule, random_state: RandomState):
+        raise TypeError("HalformIncrease would cause UBR to behave like OBR")
+
+    def centre_spread(self, rule: Rule, random_state: RandomState):
+        assert isinstance(self.sigma, np.ndarray) and self.sigma.shape[0] == 2
+        bounds = rule.match.bounds
+        bounds[:, 0] += random_state.normal(scale=self.sigma[0], size=bounds.shape[0])
+        bounds[:, 1] += halfnorm.rvs(scale=self.sigma[1] / 2, size=bounds.shape[0], random_state=random_state)
+
+    def min_percentage(self, rule: Rule, random_state: RandomState):
+        assert isinstance(self.sigma, np.ndarray) and self.sigma.shape[0] == 2
+        bounds = rule.match.bounds
+        bounds[:, 0] -= halfnorm.rvs(scale=self.sigma[0] / 2, size=bounds.shape[0], random_state=random_state)
+        bounds[:, 1] += halfnorm.rvs(scale=self.sigma[1] / 2, size=bounds.shape[0], random_state=random_state)
+
 
 class Uniform(RuleMutation):
     """Uniform noise on both bounds."""
 
     def ordered_bound(self, rule: Rule, random_state: RandomState):
         rule.match.bounds += random_state.uniform(-self.sigma, self.sigma,
-                                            size=rule.match.bounds.shape)
+                                                  size=rule.match.bounds.shape)
         rule.match.bounds = np.sort(rule.match.bounds, axis=1)
 
+    def unordered_bound(self, rule: Rule, random_state: RandomState):
+        rule.match.bounds += random_state.uniform(-self.sigma, self.sigma,
+                                                  size=rule.match.bounds.shape)
+
+    def centre_spread(self, rule: Rule, random_state: RandomState):
+        assert isinstance(self.sigma, np.ndarray) and self.sigma.shape[0] == 2
+        bounds = rule.match.bounds
+        bounds[:, 0] += random_state.uniform(-self.sigma[0], self.sigma[0], size=bounds.shape[0])
+        bounds[:, 1] += random_state.uniform(-self.sigma[1], self.sigma[1], size=bounds.shape[0])
+
+    def min_percentage(self, rule: Rule, random_state: RandomState):
+        assert isinstance(self.sigma, np.ndarray) and self.sigma.shape[0] == 2
+        bounds = rule.match.bounds
+        bounds[:, 0] += random_state.uniform(-self.sigma[0], self.sigma[0], size=bounds.shape[0])
+        bounds[:, 1] += random_state.uniform(-self.sigma[1], self.sigma[1], size=bounds.shape[0])
```

Comment:
Why is this the same as `centre_spread`?

Reply:
They both work the same in this case. Both alleles are moved using uniform noise.